pkg/trafficmngr/iptables/iptables.go

```diff
@@ -109,7 +109,7 @@ func (iptm *IPTablesManager) SetupAndEnsureMasqRules(ctx context.Context, flanne
 
 	if !flannelIPv4Net.Empty() {
 		// recycle iptables rules only when network configured or subnet leased is not equal to current one.
-		if prevNetwork != flannelIPv4Net && prevSubnet != currentlease.Subnet {
+		if !(flannelIPv4Net.Equal(prevNetwork) && prevSubnet.Equal(currentlease.Subnet)) {
```

Comment:
Instead of "&&", shouldn't it be "||"? I understand you want to recreate the iptables rules if any changes

Reply:
that's what it should do because there's the negation ! before

I think before it was wrong.
It can be either `! (a && b)` or `!a || !b`.
I can change it to the second option if you find it more readable.

Reply:
you are right, sorry I thought it was negating each!